packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -73,13 +73,18 @@ function createHttpClient(): HttpClient {
 
 export function getCacheVersion(
   paths: string[],
-  compressionMethod?: CompressionMethod
+  compressionMethod?: CompressionMethod,
+  crossOsEnabled = false
 ): string {
-  const components = paths.concat(
-    !compressionMethod || compressionMethod === CompressionMethod.Gzip
-      ? []
-      : [compressionMethod]
-  )
+  const components = paths
+    .concat(
+      !compressionMethod || compressionMethod === CompressionMethod.Gzip
+        ? []
+        : [compressionMethod]
+    )
+    .concat(
+      process.platform !== 'win32' || crossOsEnabled ? [] : ['windows-only']
```

Comment:
Clever change but I am lacking to understand if this is a sufficient change? If I am not wrong, we will end up using gnu tar even when `crossOsEnabled` set to false.

Reply:
Yes. So GNU tar on windows to windows has not shown any problems. The problems encountered is mostly on cross-os. 
If we go for bsd tar then old problems would still persist. I feel we should still fix those issues by using GNU tar

Reply:
This means we would go to gnu tar + gzip, as zstd is not present on hosted windows. Isn't it? Have we tested that?

Reply:
correcttion. zstd is installed. Earlier it was explicitly ignored.

Reply:
For self hosted windows runners, yes. And we have tested that as when we try to restore old caches done using gzip, we use gnutar + gzip combination.

Reply:
> correcttion. zstd is installed. Earlier it was explicitly ignored.

Yes, for github hosted runner we have zstd

Reply:
I agree with this approach.

Reply:
@pdotl eyes from you please!

Reply:
I also agree that we should use gnutar for windows to windows. Thanks for the explicit callout here @bishal-pdMSFT.